### PR TITLE
NOX-657 add entity services

### PR DESCRIPTION
**Pull request by anna-naumova25**

### CODE REVIEW

**Review by DanielHodgeGH — DISMISSED** [review]
Thanks @anna-naumova25 looks good - the only thing for me - which we need to discuss is the GetAllAsync - I think should use SEARCHFILTER and ORDERBY options to build on the fly GET ODATA querys to filter and order the returning dataset - but this needs to be agreed for a future PR probably - thanks DanHo

File: src/Nox.Generator.Tasks/Ui/Extensions/ServiceCollectionExtension.scriban-cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using Nox.Ui.Blazor.Lib.Services;
+using {{codeGenConventions.UiNameSpace}}.Services;
+
+namespace {{codeGenConventions.UiNameSpace}}.Extensions;
+
+public static class {{className}}
+{
+    public static IServiceCollection AddNoxUi(this IServiceCollection services)
+    {
+        services.AddSingleton<GlobalDataService>();
```
Comment (rochar) [inline src/Nox.Generator.Tasks/Ui/Extensions/ServiceCollectionExtension.scriban-cs]: @anna-naumova25 lets rename GlobalDataService, I always need to open it to understand what is the goal of it :)

maybe?
```suggestion
        services.AddSingleton<ApplicationState>();
```

File: src/Nox.Generator.Tasks/Ui/Extensions/ServiceCollectionExtension.scriban-cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        services.AddMudServices();
```
Comment (rochar) [inline src/Nox.Generator.Tasks/Ui/Extensions/ServiceCollectionExtension.scriban-cs:11]: @anna-naumova25  can you take a look at this https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests

and review if we can make this more resilient?

File: samples/Cryptocash.Ui/Nox.Generated/Ui.Extensions/ServiceCollectionExtension.g.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+namespace Cryptocash.Ui.Extensions;
+
+public static class ServiceCollectionExtension
+{
+    public static IServiceCollection AddNoxUi(this IServiceCollection services)
+    {
+        services.AddSingleton<GlobalDataService>();
+        services.AddMudServices();
+        services.AddHttpClient();
+        services.AddScoped<EndpointsProvider>();
```
Comment (rochar) [inline samples/Cryptocash.Ui/Nox.Generated/Ui.Extensions/ServiceCollectionExtension.g.cs]: this one can be singleton? never changes?

File: samples/Cryptocash.Ui/Nox.Generated/Ui.Services/EndpointsProvider.g.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public string BaseUrl
+    {
+        get { return _baseUrl; }
+    }
+
+    public string BaseUrlWithRoutePrefix
+    {
+        get { return $"{BaseUrl}/api"; }
```
Comment (rochar) [inline samples/Cryptocash.Ui/Nox.Generated/Ui.Services/EndpointsProvider.g.cs:41]: We should consider not creating a new string every time, in this case IMO it's preferable to keep the value in Memory and return always the same instance. What do y thin?

**Review by rochar — CHANGES_REQUESTED** [review]

File: samples/Cryptocash.Ui/Nox.Generated/Ui.Services/EndpointsProvider.g.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
+﻿// Generated
+
+namespace Cryptocash.Ui.Services;
+
+public class EndpointsProvider
```
Comment (rochar) [inline samples/Cryptocash.Ui/Nox.Generated/Ui.Services/EndpointsProvider.g.cs]: Should Abstract, and request the Interface in the DI

**Comment by anna-naumova25** [comment]
> Thanks @anna-naumova25 looks good - the only thing for me - which we need to discuss is the GetAllAsync - I think should use SEARCHFILTER and ORDERBY options to build on the fly GET ODATA querys to filter and order the returning dataset - but this needs to be agreed for a future PR probably - thanks DanHo

Let's see if we can use Simple.OData.Client
https://www.odata.org/blog/advanced-odata-tutorial-with-simple-odata-client/
https://www.nuget.org/packages/Simple.OData.Client/

File: samples/Cryptocash.Ui/Nox.Generated/Ui.Services/EndpointsProvider.g.cs (hunk 4 above)
Comment (anna-naumova25) [reply samples/Cryptocash.Ui/Nox.Generated/Ui.Services/EndpointsProvider.g.cs:41]: as discussed, will be done in scope of NOX-650

File: src/Nox.Generator.Tasks/Ui/Extensions/ServiceCollectionExtension.scriban-cs (hunk 1 above)
Comment (anna-naumova25) [reply src/Nox.Generator.Tasks/Ui/Extensions/ServiceCollectionExtension.scriban-cs]: updated

File: src/Nox.Generator.Tasks/Ui/Extensions/ServiceCollectionExtension.scriban-cs (hunk 2 above)
Comment (anna-naumova25) [reply src/Nox.Generator.Tasks/Ui/Extensions/ServiceCollectionExtension.scriban-cs:11]: updated

File: samples/Cryptocash.Ui/Nox.Generated/Ui.Extensions/ServiceCollectionExtension.g.cs (hunk 3 above)
Comment (anna-naumova25) [reply samples/Cryptocash.Ui/Nox.Generated/Ui.Extensions/ServiceCollectionExtension.g.cs]: updated

File: samples/Cryptocash.Ui/Nox.Generated/Ui.Services/EndpointsProvider.g.cs (hunk 5 above)
Comment (anna-naumova25) [reply samples/Cryptocash.Ui/Nox.Generated/Ui.Services/EndpointsProvider.g.cs]: updated

**Review by rochar — APPROVED** [review]

**Review by DanielHodgeGH — APPROVED** [review]

**Pull request merged**